pkg/resource/deploy/step_generator.go

```diff
@@ -52,7 +52,7 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, err
 
 	goal := event.Goal()
 	// generate an URN for this new resource.
-	urn := sg.generateURN(event)
+	urn := sg.plan.generateURN(goal)
```

Comment:
one of us is going to have a merge conflict with this, since I also refactored `generateURN` to be more general so it can be used with `Read`: https://github.com/pulumi/pulumi/pull/1654/files#diff-152cf3031402283e2b7aa998a44043ccR490152cf3031402283e2b7aa998a44043ccR490